src/exo/LoopIR_scheduling.py

```diff
@@ -2722,10 +2722,10 @@ def make_closure():
         # Construct the parameters and arguments
         args = []
         fnargs = []
-        for sym, typ, _ in sym_env:
+        for sym, typ, mem in sym_env:
             if sym in body_symbols:
                 args.append(LoopIR.Read(sym, [], typ, info))
-                fnargs.append(LoopIR.fnarg(sym, typ, None, info))
+                fnargs.append(LoopIR.fnarg(sym, typ, mem, info))
```

Comment:
Isn't this implicitly assigning `mem` to control types, too? I think having `mem = DRAM` on `size` and `index` don't really make sense (even though they are implicit in the LoopIR and are never printed).

Reply:
It will use whatever memory type was assigned to these `size`/`index` variables which should be `None`.

Reply:
Can you add an assert to check that it's actually None?

Reply:
Actually, currently the behavior is to set all the argument to DRAM if they don't have a memory specified from the front-end code: https://github.com/exo-lang/exo/blob/9d8e69cdaafdc45646ab592f0ed77ca9aa4719fc/src/exo/typecheck.py#L43

Reply:
It also ultimately gets suppressed at codegen: https://github.com/exo-lang/exo/blob/9d8e69cdaafdc45646ab592f0ed77ca9aa4719fc/src/exo/LoopIR_compiler.py#L538

I think it is safe to keep it as DRAM and it also means that we can relax the `mem?` in the LoopIR definition of `fnarg` to simply `mem`